java-manta-client-unshaded/src/main/java/com/joyent/manta/client/FindForkJoinPoolFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client;
+
+import com.joyent.manta.config.ConfigContext;
+import org.apache.commons.lang3.Validate;
+
+import java.util.concurrent.ForkJoinPool;
+
+/**
+ * Factory class that returns a {@link ForkJoinPool} instance configured with
+ * the proper number of maximum threads.
```

Comment:
"proper" here, "correct" below.  I think this javadoc would be more direct if it explained that the pool size is base on the number of connections, and that threads > connections is a problem.